app/javascript/article-form/components/TagsField.jsx

```diff
@@ -1,34 +1,90 @@
 import { h } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
 import PropTypes from 'prop-types';
-import { Tags } from '../../shared/components/tags';
+import { TagAutocompleteOption } from './TagAutocompleteOption';
+import { TagAutocompleteSelection } from './TagAutocompleteSelection';
+import { MultiSelectAutocomplete } from '@crayons';
+import { fetchSearch } from '@utilities/search';
 
-export const DEFAULT_TAG_FORMAT = '[0-9A-Za-z, ]+';
+/**
+ * TagsField for the article form. Allows users to search and select up to 4 tags.
+ *
+ * @param {Function} onInput Callback to sync selections to article form state
+ * @param {string} defaultValue Comma separated list of any currently selected tags
+ * @param {Function} switchHelpContext Callback to switch the help context when the field is focused
+ */
+export const TagsField = ({ onInput, defaultValue, switchHelpContext }) => {
+  const [defaultSelections, setDefaultSelections] = useState([]);
+  const [defaultsLoaded, setDefaultsLoaded] = useState(false);
+  const [topTags, setTopTags] = useState([]);
+
+  useEffect(() => {
+    window
+      .fetch('/tags/suggest')
+      .then((res) => res.json())
+      .then((results) => setTopTags(results));
+  }, []);
+
+  useEffect(() => {
+    // Previously selected tags are passed as a plain comma separated string
+    // Fetching furher tag data allows us to display a richer UI
+    // This fetch only happens once on first component load
+    if (defaultValue && defaultValue !== '' && !defaultsLoaded) {
+      const tagNames = defaultValue.split(', ');
```

Comment:
TBH I only formatted it like this because it's how we currently store the tags... We could definitely split by `,` and then strip any whitespace 🤔  Maybe it makes more sense that way?

I don't have too strong an opinion because I think eventually we need to rethink how tags are delivered with an article anyway (e.g. send the full tag info along with color etc, or at least send an array to make more sense with the new input).

Reply:
Ahh, I didn't realize the backend stored it with spaces. All good then. Not for this PR and maybe a never todo, but we probably don't need spaces between them stored in the DB, unless there's some obscure reason I'm unaware of. cc: @jeremyf @citizen428

Reply:
Yeah I'm not sure either - it's taken from `article.cached_tag_list`... I have a feeling it might just be because it's how the frontend was constructing/sending it before 😄